### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by MCUdude**
I've read through the patches provided in issue #495 and added what I think the documentation is lacking.
Please have a look!

### CODE REVIEW

**Comment by mcuee** [comment]
The changes look good to me.

@stefanrueger Maybe you want to take a look as well.

File: src/avrdude.1 [hunk 1]
```diff
@@ -371,6 +372,13 @@ or
 .Pa ${HOME}/.avrduderc
 file to assign a default value to keep from having to specify this
 option on every invocation.
+.Pp
+Note that some official Microchip programmers stores the bitclock setting and
+will continue to use it until a different value is provided. This applies to
+"2nd gen" programmers (AVRISPmkII, AVR Dragon, JTAG ICE mkII, STK600) and
+"3rd gen"programmers (JTAGICE3, Atmel ICE, Power Debugger). "4th gen"
+programmers (PICkit 4, MPLAB SNAP) will store the last user-specified bitclock
+until it's disconnected from the computer.
```
Comment (stefanrueger) [inline src/avrdude.1:381]: Small suggestions:
programmers stores -> programmers store
it's disconnected -> the programmer is disconnected

File: src/avrdude.1 [hunk 2]
```diff
@@ -1018,26 +1026,15 @@ by 1E3.
...
-Note that unlike STK500 settings, this setting will be reverted to
-its default value (approximately 1 microsecond) when the programming
-software signs off from the JTAG ICE.
-This parameter can also be used on the JTAG ICE mkII, JTAGICE3, and Atmel-ICE to specify the
-ISP clock period when operating the ICE in ISP mode.
+microseconds. Note that some official Microchip programmers stores the
+bitclock setting and will continue to use it until a diferent value is
+provided. See
+.Em -B bitclock
+for more information.
```
Comment (stefanrueger) [inline src/avrdude.1:1035]: programmers stores -> programmers store

File: src/doc/avrdude.texi [hunk 3]
```diff
@@ -476,6 +476,13 @@ can use the 'default_bitclock' keyword in your
 configuration file to assign a default value to keep from having to
 specify this option on every invocation.
 
+Note that some official Microchip programmers stores the bitclock setting and
+will continue to use it until a different value is provided. This applies to
+"2nd gen" programmers (AVRISPmkII, AVR Dragon, JTAG ICE mkII, STK600) and
+"3rd gen"programmers (JTAGICE3, Atmel ICE, Power Debugger). "4th gen"
+programmers (PICkit 4, MPLAB SNAP) will store the last user-specified bitclock
+until it's disconnected from the computer.
+
```
Comment (stefanrueger) [inline src/doc/avrdude.texi:485]: Small suggestions:
programmers stores -> programmers store
it's disconnected -> the programmer is disconnected

File: src/doc/avrdude.texi [hunk 4]
```diff
@@ -1719,21 +1726,13 @@ multiplies by 1E6, a trailing letter @code{k} by 1E3.
...
-@emph{JTAG ICE only:}
-Set the JTAG ICE bit clock period to @var{period} microseconds.
-Note that unlike STK500 settings, this setting will be reverted to
-its default value (approximately 1 microsecond) when the programming
-software signs off from the JTAG ICE.
-This parameter can also be used on the JTAG ICE mkII/3 to specify the
-ISP clock period when operating the ICE in ISP mode.
+Note that some official Microchip programmers stores the bitclock setting and
+will continue to use it until a diferent value is provided. See
+@code{-B bitclock} for more information.
```
Comment (stefanrueger) [inline src/doc/avrdude.texi:1732]: programmers stores -> programmers store

**Review by stefanrueger — APPROVED** [review]
Good improvement of the docs. Just tiny changes suggested

**Comment by MCUdude** [comment]
PR should be ready to be merged!

**Pull request merged**